test/NetTopologySuite.Samples.Console/Technique/SplitPolygonAtDateline.cs

```diff
@@ -0,0 +1,802 @@
+                var pRingShifted = new LinearRing(coordsShifted.ToArray());
+                var pPolygonShifted = new Polygon(pRingShifted);
+                DebugPrintCoords(pPolygonShifted.Coordinates.ToList(), "pPolygonShifted count");
+                var valOp = new IsValidOp(pPolygonShifted);
+                Debug.Print("pPolygonShifted isValid = {0}", valOp.IsValid);
+
+                // >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+                // now we determine if we will cut the polygon with a cutter line or a cutter polygon.
+                // a zero width trim gap is interpreted as a cutter line.
+                // >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+                Geometry nodedLinework = null;
+                if (double.IsInfinity(outTrimGap) || double.IsNaN(outTrimGap) || outTrimGap == 0d)
+                {
+                    // $$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$
+                    // use a cutter line to split the main polygon
+                    // $$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$
+
+                    // define the cutter line that is based on the dateline with a shift to put it in the same hemisphere as the shifted
+                    // polygon.
+                    var cutterLine = DatelineCutterPolyline(delNeg, delPos, shift);
+                    DebugPrintCoords(cutterLine.Coordinates.ToList(), "cutterLine count");
+
+                    // perform a Union to split the polygon with a polyline
+                    nodedLinework = pPolygonShifted.Boundary.Union(cutterLine);
+                }
+                else
+                {
+                    // $$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$
+                    // use a cutter polygon to split the main polygon
+                    // $$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$
+
+                    // define the cutter polygon that is based on the dateline with a shift to put it in the same hemisphere as the shifted
+                    // polygon. Plus we also add the trim gap as the cutter polygon width.
+                    var cutterPgon = DatelineCutterPolygon(delNeg, delPos, shift, outTrimGap);
+                    DebugPrintCoords(cutterPgon.Coordinates.ToList(), "cutterPgon count");
+                    var valOp2 = new IsValidOp(cutterPgon);
+                    Debug.Print("cutterPgon isValid = {0}", valOp2.IsValid);
+
+                    // perform an Erase to split the polygon with a polygon. this consists of an Intersection and then a Difference.
+                    //var intx = pPolygonShifted.Boundary.Intersection(cutterPgon);
```

Comment:
Old commented code that can be removed?

Reply:
yep. will remove after the review is done. I'm an over-debugger I have to confess. I wanted to double check every output so  have tons of debugs. :)